### PR TITLE
Begin to plug the profiling gaps in the \ceil simplifier

### LINKED ISSUE #1896: Profile \ceil simplifier in eventlog
We should add eventlog profiling events to the `\ceil` simplifier to make this information available to users. We would also like this information to investigate alternative strategies for managing `\ceil` conditions.

### PR DESCRIPTION
**Pull request by ttuegel**
Fixes #1896 

---

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Step/Simplification/Equals.hs [hunk 1]
```diff
@@ -58,8 +58,7 @@ import Kore.Step.Simplification.AndTerms
     ( maybeTermEquals
     )
 import {-# SOURCE #-} qualified Kore.Step.Simplification.Ceil as Ceil
-    ( makeEvaluate
-    , makeEvaluateTerm
+    ( makeEvaluateTerm
```
Comment (ana-pantilie) [inline kore/src/Kore/Step/Simplification/Equals.hs:61]: Why is this still needed?

File: kore/src/Kore/Step/Simplification/CeilSimplifier.hs [hunk 2]
```diff
@@ -105,3 +121,15 @@ runCeilSimplifierWith
...
 {-# INLINE runCeilSimplifierWith #-}
+
+ceilSimplifierTermLike
+    ::  InternalVariable variable
+    =>  MonadSimplify simplifier
+    =>  CeilSimplifier
+            (ReaderT (SideCondition variable) simplifier)
+            (TermLike variable)
+            (OrCondition variable)
+ceilSimplifierTermLike =
```
Comment (ana-pantilie) [inline kore/src/Kore/Step/Simplification/CeilSimplifier.hs:132]: Where is this used, or where/why will this be needed?

Comment (ttuegel) [reply kore/src/Kore/Step/Simplification/CeilSimplifier.hs:132]: It's not, thanks for spotting that!

File: kore/src/Kore/Step/Simplification/Equals.hs (hunk 1 above)
Comment (ttuegel) [reply kore/src/Kore/Step/Simplification/Equals.hs:61]: It's not either, and if I also remove the import from `Kore.Builtin.Builtin`, then we can remove the `.hs-boot` file for `Ceil`! :tada:

**Comment by ttuegel** [comment]
> I couldn't find anything else to refactor, except removing some outdated comments.

That sounds great! "Cleanup" might be better than "refactor" anyway. I think cleanup can include refactoring.

**Pull request merged**